src/collections/_documentation/platforms/python/troubleshooting.md

```diff
@@ -1,19 +1,58 @@
 ---
-title: Contextvars vs thread locals
+title: Troubleshooting
 sidebar_order: 11
 ---
 
-The Python SDK generally does its best to figure out how contextual data such
-as tags set with `sentry_sdk.set_tags` is supposed to flow along your control
-flow. In most usecases this is achieved perfectly fine using [thread
-locals](https://docs.python.org/3/library/threading.html#thread-local-data),
-but there are quite a few situations where this approach fails.
+This section contains various gotchas that are inherent to the Python SDK's
+design, do not necessarily affect the majority of users and do not really fit
+anywhere else.
 
-Read this document if you cannot figure out why context data is leaking across
-your HTTP requests, or when data is missing or popping up at the wrong place
-and time.
+The information here will help you answer these questions:
 
-## Python 2: thread locals and gevent
+- "What do I do if scope data is leaking between requests?"
+- "What do I do if my transaction has nested spans when they should be parallel?"
+
+The short answer to those: check if your `contextvars` work and you clone the
+hub where needed.
+
+## Addressing concurrency issues
+
+Python supports several distinct solutions to concurrency, including threads and
+coroutines.
+
+The Python SDK does its best to figure out how contextual data such as tags set
+with `sentry_sdk.set_tags` is supposed to flow along your control flow. In most
+cases it works perfectly, but in a few situations some special care must be
+taken. This is specially true when working with a code base doing concurrency
+outside of the provided framework integrations.
+
+The general recommendation is to have one hub per "concurrency unit"
+(thread/coroutine/etc). The SDK ensures every thread has an independent hub. If
+you do concurrency with `asyncio` coroutines, clone the current hub for use
+within a block that runs concurrent code:
+
+```python
+with Hub(Hub.current):
+    # in this block Hub.current refers to a new clone
+    # of the original hub, with the same client and
+    # the same initial scope data.
+```
+
+Issues with `asyncio` have then an easy workaround: every `asyncio` coroutine
+that really does run concurrently with other coroutines needs to be made into a
+task, then the hub needs to be cloned and reassigned.
```

Comment:
```suggestion
task, then the hub needs to be cloned as shown above.
```

Reply:
Thanks. I was not sure what you meant with "reassigned" originally, but now I see that `Hub.__enter__` sets the current hub, so that should be enough.

I wanted to add a more complete example, similar to what is in https://github.com/getsentry/sentry-python/issues/147 but with your suggested correction applied. WDYT?